controllers/credential_controller.go

```diff
@@ -2,34 +2,268 @@ package controllers
 
 import (
 	"context"
-
+	"fmt"
+	civ1 "github.com/estafette/estafette-ci-operator/api/v1"
 	"github.com/go-logr/logr"
+	yaml "gopkg.in/yaml.v2"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+)
 
-	civ1 "github.com/estafette/estafette-ci-operator/api/v1"
+var (
+	configMapOwnerKey = ".metadata.controller"
+	configMapName     = "estafette-external-credentials"
+	apiGVStr          = civ1.GroupVersion.String()
 )
 
 // CredentialReconciler reconciles a Credential object
 type CredentialReconciler struct {
 	client.Client
-	Log logr.Logger
+	Log      logr.Logger
+	Scheme   *runtime.Scheme
+	Recorder record.EventRecorder
 }
 
 // +kubebuilder:rbac:groups=ci.estafette.io,resources=credentials,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=ci.estafette.io,resources=credentials/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=core,resources=configmaps,verbs=get;list;create;update;patch;delete
 
 func (r *CredentialReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
-	_ = context.Background()
-	_ = r.Log.WithValues("credential", req.NamespacedName)
+	ctx := context.Background()
```

Comment:
Doesn't the incoming `req` argument have a context we can reuse?

Reply:
It doesn't. Is not an http request, is controller reconcile request: https://godoc.org/sigs.k8s.io/controller-runtime/pkg/reconcile#Request

Plus, this comes from recommendation of controller implementation.